packages/vite/src/node/optimizer/index.ts

```diff
@@ -233,20 +252,25 @@ export async function optimizeDeps(
 
   const start = Date.now()
 
+  const { entryPoints = [], plugins = [], ...esbuildOptions } =
+    config.optimizeDeps?.esbuildOptions ?? {}
+
   const result = await build({
-    entryPoints: Object.keys(flatIdDeps),
+    entryPoints: [...entryPoints, ...Object.keys(flatIdDeps)],
```

Comment:
Does it make sense to merge the user's `entryPoints`? Or should we simply add it to the excluded settings?

Reply:
What would be the use case of adding extra `entryPoints` here? There is logic needed to flat ids that wouldn't be applied if the user has control of them. Maybe we should also omit this option (it could be added later also in case we find a use case)

Reply:
Yeah, makes sense. I removed it.